app/routes/events/components/StripeElement.js

```diff
@@ -4,38 +4,146 @@ import {
   injectStripe,
   PaymentRequestButtonElement,
   Elements,
-  StripeProvider
+  StripeProvider,
+  CardCVCElement,
+  CardExpiryElement,
+  CardNumberElement
 } from 'react-stripe-elements';
 import config from 'app/config';
-import StripeCheckout from 'react-stripe-checkout';
-import Button from 'app/components/Button';
-import styles from './StripeElement.css';
-import logoImage from 'app/assets/kule.png';
+import stripeStyles from './Stripe.css';
 import type { EventRegistrationChargeStatus, User, Event } from 'app/models';
+import LoadingIndicator from 'app/components/LoadingIndicator';
 
-import { paymentPending } from '../utils';
+type Stripe = {
+  paymentRequest: Object => Object,
+  handleCardPayment: string => Promise<*>,
+  confirmPaymentIntent: (string, Object) => Promise<*>
+};
 
 type Props = {
   event: Event,
   currentUser: User,
-  onToken: (token: string) => Promise<*>,
+  createPaymentIntent: () => Promise<*>,
   chargeStatus: EventRegistrationChargeStatus
 };
 
-type FormProps = Props & { stripe: { paymentRequest: Object => Object } };
+type StripeError = {
+  type: string,
+  code: string,
+  message: string,
+  doc_url: string
+};
+
+type FormProps = Props & {
+  fontSize?: number
+};
+
+type _SplitFormProps = FormProps & {
+  ledgend: string,
+  setError: Object => void,
+  setSuccess: () => void,
+  setLoading: boolean => void,
+  stripe: Stripe
+};
+
+type _PaymentRequestFormProps = FormProps & {
+  setError: Object => void,
+  setSuccess: () => void,
+  setLoading: boolean => void,
+  setPaymentRequest: boolean => void,
+  stripe: Stripe
+};
+
+type FormState = {
+  error?: StripeError | null,
+  success?: boolean,
+  loading: boolean,
+  paymentRequest: boolean
+};
 
 type State = {
-  paymentRequest: Object,
+  paymentRequest?: Object,
   canMakePayment?: boolean
 };
+const createOptions = (fontSize, padding) => {
```

Comment:
`CreateOptions`-whaaat?

Also `(fntS, padd) => ({ style: ...})`

Reply:
I'll just make this an object. there's no need to have it as a function.